micro-elasticache/src/main/java/com/aol/micro/server/elasticache/manifest/comparator/ElasticacheManifestComparator.java

```diff
@@ -0,0 +1,280 @@
+package com.aol.micro.server.elasticache.manifest.comparator;
+
+import com.aol.cyclops2.util.ExceptionSoftener;
+import com.aol.micro.server.distributed.DistributedCache;
+import com.aol.micro.server.distributed.DistributedMap;
+import com.aol.micro.server.manifest.Data;
+import com.aol.micro.server.manifest.ManifestComparator;
+import com.aol.micro.server.manifest.ManifestComparatorKeyNotFoundException;
+import com.aol.micro.server.manifest.VersionedKey;
+import com.aol.micro.server.rest.jackson.JacksonUtil;
+import cyclops.control.Xor;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Optional;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class ElasticacheManifestComparator<T> implements ManifestComparator<T> {
+
+    /**
+     * Manifest comparator for use with a distributed map -assumes single producer /
+     * multiple consumers
+     *
+     * Uses to entries in the map
+     *
+     * key : versioned key versioned key : actual data
+     *
+     * ManifestComparator stores the current version number, only when the version
+     * changes is the full data set loaded from the remote store.
+     *
+     * Usage as a Spring Bean - inject into the host class, and use withKey to
+     * customise for the targeted Key.
+     *
+     *
+     * <pre>
+     * {@code
+     * &#64;Rest
+    public class MyDataService {
+
+
+
+    private final ManifestComparator<DataType> comparator;
+    &#64;Autowired
+    public  MyDataService(ManifestComparator comparator) {
+    this.comparator = comparator.withKey("test-key");
+    }
+     *
+     * }
+     * </pre>
+     *
+     * micro-elasticache configures a single ManifestComparator bean that can be
+     * customized for multiple different keys via withKey
+     *
+     * When your bean is injected save via saveAndIncrement, and periodically call
+     * load() to refresh data if (and only if) it has changed.
+     *
+     * ManifestComparator will automatically remove old versions on
+     * saveAndIncrement, but system outages may occasionally cause old keys to
+     * linger, you can also use clean & cleanAll to periodically to remove old key
+     * versions.
+     *
+     *
+     * @author gordonmorrow
+     *
+     * @param <T>
+     */
+
+        private final Logger logger = LoggerFactory.getLogger(getClass());
+
+        private final String key;
+
+        private volatile Xor<Void, T> data = Xor.secondary(null); // Void represents
```

Comment:
Why `Xor<Void, T>` over just `Optional<T>`? Or if you're worried about tracking presence or not, `Future<Optional<T>>` ?